tests/test_api.py

```diff
@@ -357,3 +359,23 @@ def test_export_cross_prediction():
     )
 
     assert isinstance(sorted_dict, dict)
+
+
+@pytest.mark.parametrize("test_df", test_dfs)
+def test_allocation_limit(test_df):
+    """Test used to see if calculated allocation values are inside of specified limit"""
+
+    test_df_copy = copy.deepcopy(test_df)
+    df_with_allocations = Api.calculate_allocations(
+        df=test_df_copy, name_of_strategy=available_allocation_algorithms[0], name_of_price_series="close",
+        allocation_lower_limit=0, allocation_upper_limit=0
+    )
+    if all(df_with_allocations["allocation"] != 0):
+        raise ValueError("Allocation limits breached")
+
+    df_with_allocations = Api.calculate_allocations(
+        df=test_df_copy, name_of_strategy=available_allocation_algorithms[0], name_of_price_series="close",
+        allocation_lower_limit=-0.1, allocation_upper_limit=0.1
+    )
+    if all(-0.1 > df_with_allocations["allocation"]) or all(df_with_allocations["allocation"] > 0.1):
```

Comment:
```suggestion
    if any(-0.1 > df_with_allocations["allocation"]) or any(df_with_allocations["allocation"] > 0.1):
```

Reply:
@NikolaR01 - any rather than all to catch any exception to the limits?

Reply:
Sure, but I think it would work the same since in both cases the statement would be false in case of one value being out of bounds